scripts/install

```diff
@@ -10,198 +10,198 @@ source _common.sh
-# DEPENDENCIES
+# CREATE DEDICATED USER
 #=================================================
-# Activate backports sources.list
-cp -a "../conf/turtl.list" "/etc/apt/sources.list.d/$app.list"
-cp -a "../conf/turtl-preferences" "/etc/apt/preferences.d/00TurtlPinning"
-if [ $archi == "armv7l" ]
-then
-    gpg --list-keys 7638D0442B90D010 > /dev/null 2>&1
-    if [ $? != 0 ]
-    then
-        gpg --keyserver pgpkeys.mit.edu --recv-key 7638D0442B90D010
-    fi
-    gpg --list-keys 8B48AD6246925553 > /dev/null 2>&1
-    if [ $? != 0 ]
-    then
-        gpg --keyserver pgpkeys.mit.edu --recv-key 8B48AD6246925553
-    fi
-    if [ $(apt-key finger | grep "7638 D044 2B90 D010" -c) == 0 ]
-    then
-        gpg -a --export 7638D0442B90D010 | apt-key add -
-    fi
-    if [ $(apt-key finger | grep "8B48 AD62 4692 5553" -c) == 0 ]
-    then
-        gpg -a --export 8B48AD6246925553 | apt-key add -
-    fi
-fi
+ynh_script_progression --message="Configuring system user..." --weight=1
 
-if [ $archi == "armv7l" ]
-then
-    # Install RethinkDB
-    dpkg -i ../conf/rethinkdb_2.3.6_armhf.deb
-else
-    # Activate RethinkDB sources.list
-    release=$(lsb_release -cs)
-    echo "deb http://download.rethinkdb.com/apt $release main" | tee /etc/apt/sources.list.d/rethinkdb.list
-    wget -qO- https://download.rethinkdb.com/apt/pubkey.gpg | apt-key add -
-    ynh_package_update
-fi
+# Create a system user
+ynh_system_user_create --username=$app --home_dir="$final_path"
 
-# Install dependencies
-ynh_package_update
-ynh_install_app_dependencies wget git build-essential rethinkdb libuv1-dev python-pip
+#=================================================
+# CREATE A POSTGRESQL DATABASE
+#=================================================
+ynh_script_progression --message="Creating a PostgreSQL database..." --weight=2
 
+db_name=$(ynh_sanitize_dbid --db_name=$app)
+db_user=$db_name
+ynh_app_setting_set --app=$app --key=db_name --value=$db_name
+ynh_psql_test_if_first_run
+ynh_psql_setup_db --db_user=$db_name --db_name=$db_name
 
-# Install Clozure Common Lisp
-cd /opt
-if [ $archi == "armv7l" ]
-then
-    wget -q ftp://ftp.clozure.com/pub/release/1.11/ccl-1.11-linuxarm.tar.gz
-    tar xf ccl-1.11-linuxarm.tar.gz
-else
-    wget -q ftp://ftp.clozure.com/pub/release/1.11/ccl-1.11-linuxx86.tar.gz
-    tar xf ccl-1.11-linuxx86.tar.gz
-fi
+#=================================================
+# DOWNLOAD, CHECK AND UNPACK SOURCE
+#=================================================
+ynh_script_progression --message="Setting up source files..." --weight=3
 
-cd ccl
-if [ $(grep -c "flags.* lm .*" /proc/cpuinfo) -eq 0 ]
-then
-    cp scripts/ccl /usr/bin/ccl
-else
-    cp scripts/ccl64 /usr/bin/ccl
-fi
-sed -e "s@CCL_DEFAULT_DIRECTORY=/usr/local/src/ccl@CCL_DEFAULT_DIRECTORY=/opt/ccl@" -i /usr/bin/ccl
+ynh_app_setting_set --app=$app --key=final_path --value=$final_path
+git clone https://github.com/turtl/server "$final_path" --quiet
```

Comment:
Zblerg, I don't know why the new trend is to use git clones more and more, is there any reason for this ?

Reply:
changed to the usual app.src